server/service/osquery.go

```diff
@@ -1456,27 +1460,26 @@ func (svc *Service) SubmitResultLogs(ctx context.Context, logs []json.RawMessage
 	// otherwise the results will never clear from its local DB and
 	// will keep retrying forever.
 	//
-	unmarshaledResults, queriesDBData := svc.preProcessOsqueryResults(ctx, logs)
-	svc.saveResultLogsToQueryReports(ctx, unmarshaledResults, queriesDBData)
 
-	var filteredLogs []json.RawMessage
-	for i, unmarshaledResult := range unmarshaledResults {
-		if unmarshaledResult == nil {
-			// Ignore results that could not be unmarshaled.
-			continue
-		}
-		dbQuery, ok := queriesDBData[unmarshaledResult.QueryName]
-		if !ok {
-			// Ignore results for unknown queries.
-			continue
-		}
-		if !dbQuery.AutomationsEnabled {
-			// Ignore results for queries that have automations disabled.
-			continue
-		}
-		filteredLogs = append(filteredLogs, logs[i])
+	var queryReportsDisabled bool
+	appConfig, err := svc.ds.AppConfig(ctx)
+	if err != nil {
+		level.Error(svc.logger).Log("msg", "getting app config", "err", err)
+		// If we fail to load the app config we assume the flag to be disabled
+		// to not perform extra processing in that scenario.
+		queryReportsDisabled = true
+	} else {
+		queryReportsDisabled = appConfig.ServerSettings.QueryReportsDisabled
 	}
-	if err := svc.osqueryLogWriter.Result.Write(ctx, filteredLogs); err != nil {
+
+	unmarshaledResults, queriesDBData := svc.preProcessOsqueryResults(ctx, logs, queryReportsDisabled)
+	svc.saveResultLogsToQueryReports(ctx, unmarshaledResults, queriesDBData, queryReportsDisabled)
```

Comment:
do we need to call either of these if queryReportDisabled == true?

Reply:
Yeah, could probably skip the call, both are no-op now if `queryReportDisabled == true`.